### PR TITLE
PR: Use '!' for Pdb commands

### PR DESCRIPTION
**Pull request by impact27**
Use '!' before Pdb commands and nothing before python code, as is the case in IPython

### CODE REVIEW

File: spyder_kernels/console/kernel.py [hunk 1]
```diff
@@ -293,6 +294,13 @@ def set_pdb_execute_events(self, state):
         if self._pdb_obj:
             self._pdb_obj.pdb_execute_events = state
 
+    def set_pdb_use_exclamation_mark(self, state):
+        """
+        Handle a message from the frontend
```
Comment (ccordoba12) [inline spyder_kernels/console/kernel.py:299]: Please change this docstring to mention exactly what this method does.

File: spyder_kernels/customize/spyderpdb.py [hunk 2]
```diff
@@ -367,6 +466,24 @@ def _cmdloop(self):
                         "Could not send debugging state to the frontend.")
                 raise
 
+    def precmd(self, line):
+        """
+        Hook method executed just before the command line is
+        interpreted, but after the input prompt is generated and issued.
+
+        Here we switch ! and non !
+
```
Comment (ccordoba12) [inline spyder_kernels/customize/spyderpdb.py:475]: Please remove this blank line.

**Review by ccordoba12 — COMMENTED** [review]
I left two minor comments, otherwise looks good.

File: spyder_kernels/console/kernel.py [hunk 3]
```diff
@@ -293,6 +294,14 @@ def set_pdb_execute_events(self, state):
         if self._pdb_obj:
             self._pdb_obj.pdb_execute_events = state
 
+    def set_pdb_use_exclamation_mark(self, state):
+        """
+        Sets an option on the current debugging session to decide wether
```
Comment (ccordoba12) [inline spyder_kernels/console/kernel.py:299]: Sets -> Set

**Comment by ccordoba12** [comment]
@impact27, please remove from this PR all commits that you created for PR #251, to not mess with the history of this repo. I think there are only two: 3c9e5b0 and 04a05bf. Additionally, please also remove this merge: e1af7e0.

**Comment by ccordoba12** [comment]
Hey @impact27, is this one ready to be merged?

**Comment by impact27** [comment]
> Hey @impact27, is this one ready to be merged?

yes

**Review by ccordoba12 — APPROVED** [review]
Thanks @impact27, looks good to me now.

**Pull request merged**